src/libcore/char.rs

```diff
@@ -380,4 +378,13 @@ impl Iterator for EscapeDefault {
             EscapeDefaultState::Unicode(ref mut iter) => iter.next()
         }
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        match self.state {
+            EscapeDefaultState::Char(_) => (1, Some(1)),
+            EscapeDefaultState::Backslash(_) => (2, Some(2)),
+            EscapeDefaultState::Unicode(_) => (0, Some(10)),
+            _ => (0, Some(0))
```

Comment:
Would it be better to have `EscapeDefaultState::Done` here?
It would make it easier to understand why `(0, Some(0))` is the correct return value and in the unlikely case that the enum is extended it would promptly point out the problem.